### PR TITLE
v9: Don't load all translation files

### PR DESCRIPTION
**Pull request by Zeegaan**
Fixes https://github.com/umbraco/Umbraco-CMS/issues/11438

# Notes
- Added lazy loading to translation files so we don't load all of them when only using 1 language
- Fixed unit tests

I've posted an image here where you see everytime we load in a language we use more memory, here 21 files are loaded in: 
image

In the new one, all those are gone (only 1 language is loaded) 
image

# How to test
- Run the umbraco installer and close the website afterwards
- Set a breakpoint on line 118 in LocalizedTextServiceFileSources like so:
image
- Run Umbraco again, this should only load the languages you are using(Before this would load all languages)
- Go to user settings and change your language
- We should again only be loading the language you selected

### CODE REVIEW

File: src/Umbraco.Infrastructure/Services/Implement/LocalizedTextService.cs [hunk 1]
```diff
@@ -8,63 +8,86 @@
...
+
+        private IDictionary<CultureInfo, Lazy<IDictionary<string, string>>> _noAreaDictionarySource =>
+            _noAreaDictionarySourceLazy.Value;
+
+        private readonly Lazy<IDictionary<CultureInfo, Lazy<IDictionary<string, IDictionary<string, string>>>>>
+            _dictionarySourceLazy;
+
+        private readonly Lazy<IDictionary<CultureInfo, Lazy<IDictionary<string, string>>>> _noAreaDictionarySourceLazy;
+        private readonly char[] _splitter = new[] {'/'};
+
```
Comment (bjarnef) [inline src/Umbraco.Infrastructure/Services/Implement/LocalizedTextService.cs:28]: @Zeegaan I think we should use the `CharArrays.ForwardSlash` constant instead :)
https://github.com/umbraco/Umbraco-CMS/blob/v9/contrib/src/Umbraco.Core/Constants-CharArrays.cs#L10-L13

File: src/Umbraco.Infrastructure/Services/Implement/LocalizedTextService.cs [hunk 2]
```diff
@@ -127,7 +169,9 @@ public string Localize(string key, CultureInfo culture, IDictionary<string, stri
             var alias = keyParts.Length > 1 ? keyParts[1] : keyParts[0];
```
Comment (bjarnef) [inline src/Umbraco.Infrastructure/Services/Implement/LocalizedTextService.cs:169]: Change this part:
```
var keyParts = key.Split(_splitter, StringSplitOptions.RemoveEmptyEntries);
```
to
```
var keyParts = key.Split(CharArrays.ForwardSlash, StringSplitOptions.RemoveEmptyEntries);
```

File: src/Umbraco.Infrastructure/Services/Implement/LocalizedTextService.cs (hunk 1 above)
Comment (Zeegaan) [reply src/Umbraco.Infrastructure/Services/Implement/LocalizedTextService.cs:28]: I agree, nice catch 💪

**Comment by bergmania** [comment]
Cherry picked for 9.1

**Pull request merged**